sdk/js/src/service/devices/split.js

```diff
@@ -138,6 +138,50 @@ export async function makeRequests(
     }
   }
 
+  // Split the end device payload per component.
+  function splitPayload(payload, base = {}, direction, component) {
+    const { end_device } = payload
+
+    if (!Boolean(end_device)) {
+      return {}
+    }
+
+    const paths = Marshaler.pathsFromPatch(end_device)
+    const device = traverse(end_device)
+    const result = traverse(base)
+    const retrieveIndex = direction === 'get' ? 0 : 1
+
+    for (const path of paths) {
+      // Look up the current path in the device entity map
+      const subtree =
+        traverse(deviceEntityMap).get(path) || traverse(deviceEntityMap).get([path[0]])
+
+      if (!subtree) {
+        throw new Error(`Invalid or unknown field mask path used: ${path}`)
+      }
+
+      const definition = '_root' in subtree ? subtree._root[retrieveIndex] : subtree[retrieveIndex]
+
+      const map = function(currentComponent, path) {
+        if (component === currentComponent) {
+          result.set(path, device.get(path))
+        }
+      }
+
+      if (definition) {
+        if (definition instanceof Array) {
+          for (const comp of definition) {
+            map(comp, path)
+          }
+        } else {
+          map(definition, path)
+        }
+      }
+    }
+
+    return Marshaler.payload(result.value, 'end_device')
+  }
```

Comment:
Not sure if I miss something here, but couldn't you just compose the filtered payload using the paths array inside the `requestTree`? Like so:
```js
  function splitPayload(payload, paths) {
    if (Object.keys(payload).length === 0) return {}

    const { end_device } = payload

    const result = traverse({})
    const endDevice = traverse(end_device)

    for (const path of paths) {
      result.set(path, endDevice.get(path))
    }

    return Marshaler.payload(result.value, 'end_device')
  }
```

and then use it like e.g.:
```js
splitPayload(payload, requestTree.is)
```